gems/pending/postgres_ha_admin/database_yml.rb

```diff
@@ -0,0 +1,68 @@
+require 'util/miq-password'
+
+module PostgresHaAdmin
+  class DatabaseYml
+    attr_reader :db_yml_file, :environment
+
+    def initialize(db_yml_file, environment)
+      @db_yml_file = db_yml_file
+      @environment = environment
+    end
+
+    def pg_params_from_database_yml
+      rails_params_to_pg(params_from_database_yml)
+    end
+
+    def update_database_yml(params)
+      db_yml = YAML.load_file(db_yml_file)
+      db_yml[environment] = params_from_database_yml
+      db_yml[environment].merge!(pg_parameters_to_rails(params))
+      remove_empty(db_yml[environment])
+
+      new_name = "#{db_yml_file}_#{Time.current.strftime("%d-%B-%Y_%H.%M.%S")}"
+      File.rename(db_yml_file, new_name)
+
+      File.write(db_yml_file, db_yml.to_yaml)
+      new_name
+    end
+
+    private
+
+    def rails_params_to_pg(params)
+      pg_params = {}
+      pg_params[:dbname] = params['database']
+      pg_params[:user] = params['username']
+      pg_params[:port] = params['port']
+      pg_params[:host] = params['host']
+      pg_params[:password] = MiqPassword.try_decrypt(params['password'])
+      remove_empty(pg_params)
+    end
+
+    def pg_parameters_to_rails(pg_params)
+      params = {}
+      params['username'] = pg_params[:user]
+      params['database'] = pg_params[:dbname]
+      params['port'] = pg_params[:port]
+      params['host'] = pg_params[:host]
+      remove_empty(params)
+    end
+
+    def remove_empty(hash)
+      hash.delete_if { |_k, v| empty?(v) }
+    end
+
+    def empty?(value)
+      true if value.nil? || value.to_s.strip.empty?
+    end
+
+    def params_from_database_yml
+      yml_params = YAML.load_file(db_yml_file)
+      result = if yml_params['base'].nil?
+                 yml_params[environment]
+               else
+                 yml_params['base'].merge(yml_params[environment])
```

Comment:
I don't think this is what we always want to do.
For example, if I have a yml file that looks like this:
```yml
---
thing: &thing
  key1: value
  key2: other_value

second_thing:
  <<: *thing
  key3: second_thing value

third_thing:
  key3: third_thing value
```

I don't need to explicitly merge `thing` and `second_thing` and I wouldn't expect any of `thing`'s keys to be present in `third_thing`. So when I do a `YAML.load_file` I get:

```ruby
{
  "thing"=> {
    "key1"=>"value",
    "key2"=>"other_value"
  },
  "second_thing"=>{
    "key1"=>"value",
    "key2"=>"other_value",
    "key3"=>"second_thing value"
  },
  "third_thing"=>{
    "key3"=>"third_thing value"
  }
}
```

So, because of this behavior I wouldn't expect you to be explicitly including the `base` key here.